sqllogictest/src/parser.rs

```diff
@@ -700,4 +761,81 @@ mod tests {
             })
         }
     }
+
+    /// Replaces the actual filename in all Records with
+    /// "__FILENAME__" so different files with the same contents can
+    /// compare equal
+    fn normalize_filename(records: Vec<Record>) -> Vec<Record> {
```

Comment:
I am also not a super fan of this patter either, but it is what I can come up with

Reply:
What about using `derivative` crate to ignore a field when comparing? https://mcarton.github.io/rust-derivative/latest/cmp.html#ignoring-a-field